base/bucket_gocb_test.go

```diff
@@ -2550,3 +2550,74 @@ func TestGetStatsVbSeqNo(t *testing.T) {
 		assert.True(t, len(highSeqNos) > 0)
 	})
 }
+
+// Confirm that GoCBv2 preserveExpiry option works correctly for bucket Set function
+func TestUpsertOptionPreserveExpiry(t *testing.T) {
+	if !TestUseCouchbaseServer() {
+		t.Skip("Test can only be ran against CBS due to GoCB v2 use")
+	}
+	defer SetUpTestLogging(LevelInfo, KeyAll)()
+
+	testCases := []struct {
+		name          string
+		upsertOptions *sgbucket.UpsertOptions
+		expectMatch   bool
+	}{
+		{
+			name:          "Expect matching expiry - preserveExpiry",
+			upsertOptions: &sgbucket.UpsertOptions{PreserveExpiry: true},
+			expectMatch:   true,
+		},
+		{
+			name:          "Expect updated expiry - false preserveExpiry",
+			upsertOptions: &sgbucket.UpsertOptions{PreserveExpiry: false},
+			expectMatch:   false,
+		},
+		{
+			name:          "Expect updated expiry - nil upsert options",
+			upsertOptions: nil,
+			expectMatch:   false,
+		},
+	}
+
+	for i, test := range testCases {
+		bucket := GetTestBucketForDriver(t, GoCBv2)
+		if !bucket.IsSupported(sgbucket.DataStoreFeaturePreserveExpiry) {
+			bucket.Close()
+			t.Skip("Preserve expiry is not supported with this CBS version. Skipping entire test...")
+		}
```

Comment:
Would be nice to move this up out of the subtests for a couple of reasons:
- Only 1 test bucket for all sub tests (as long as doc IDs are unique)
- Earlier exit/skip for the non-supported CB version before we fall into the sub-tests